### PR TITLE
Add default visibility to class LegacyOpExtension

### PR DESCRIPTION
**Pull request by artkuli**
### Tickets:
 - *93540*

### CODE REVIEW

File: src/core/dev_api/legacy_op_extension.hpp [hunk 1]
```diff
@@ -9,5 +9,5 @@
 namespace ov {
 
 /** @brief Class to distinguish legacy extension. */
-class LegacyOpExtension : public BaseOpExtension {};
+class OPENVINO_API LegacyOpExtension : public BaseOpExtension {};
```
Comment (ilya-lavrenov) [inline src/core/dev_api/legacy_op_extension.hpp:12]: what does it really fix?
If we really need this, I suggest adding one virtual method to this class and put its implementation to cpp. It helps to prevents issues with RTTI on clang

Comment (ilyachur) [reply src/core/dev_api/legacy_op_extension.hpp:12]: We can add destructor for this class

Comment (artkuli) [reply src/core/dev_api/legacy_op_extension.hpp]: Destructor added

Comment (ilya-lavrenov) [reply src/core/dev_api/legacy_op_extension.hpp]: anyway, could you please elaborate in which situation we have issues which require this fix?

Comment (artkuli) [reply src/core/dev_api/legacy_op_extension.hpp]: This PR fixes the issue on MacOS with clang where the test: https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/tests/functional/extension/extension.cpp#L275 failed. The problem occurs here: https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/frontend.cpp#L175
The dynamic cast to `LegacyOpExtension` failed (nullptr) but dynamic cast to the `BaseOpExtension` works correctly. 
On Ubuntu with gcc everything works good.

**Review by ilyachur — APPROVED** [review]

**Review by praasz — APPROVED** [review]

**Pull request merged**